src/noit_filters.c

```diff
@@ -482,7 +495,7 @@ noit_apply_filterset(const char *filterset,
       need_target = !MATCHES(target, check->target);
       need_module = !MATCHES(module, check->module);
       need_name = !MATCHES(name, check->name);
-      need_metric = !noit_apply_filterrule_metric(r, metric->metric_name, strlen(metric->metric_name), &stset, &mtset);
+      need_metric = !noit_apply_filterrule_metric(r, metric->metric_name, mlen, &stset, &mtset);
```

Comment:
IIUC, this is taking the mlen from noit_metric_parse_tags(), which is the length of the metric name without tags?

Reply:
noit_metric_parse_tags() returns the input len passed into it, which is `strlen(metric->metric_name)`.  It's the same value, we just aren't recomputing it a second time.

Reply:
...and noit_apply_filterrule_metric() already supported stream tags `stsearch`

Reply:
noit_metric_parse_tags returns the length without andy tags.  The "eat_up_tags" function in there modifies (down) input_len as it ... eats up tags.

Reply:
Ah, i missed the `eat_up_tags(input, &input_len`